youtokentome/cpp/yttm.pyx

```diff
@@ -84,38 +84,40 @@ cdef class BPE:
         if status.code != 0:
             raise ValueError(status.message.decode())
 
-    def encode(self, sentences, output_type, bos, eos, reverse):
+    def encode(self, sentences, output_type, bos, eos, reverse, dropout_prob):
         cdef vector[string] s
         cdef vector[vector[string]] ret_subwords
         cdef vector[vector[int]] ret_ids
         cdef Status status
+        if dropout_prob < 0 or dropout_prob > 1:
+            raise ValueError("dropout_prob value must be in the range [0, 1]. Current value of dropout_prob = " + str(dropout_prob))
         if output_type == 'id':
             if isinstance(sentences, str):
                 s = [sentences.encode()]
                 assert len(s) == 1
```

Comment:
Could you please replace asserts without error information with `if` block, which throws an exception?

Reply:
Looks like this assert is always true. I removed it.